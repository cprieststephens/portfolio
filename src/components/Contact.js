import React from "react";
import { Link } from "react-router-dom";
import emailIcon from "../img/email-icon.png";
import linkedInIcon from "../img/linkedin-icon.png";
import xIcon from "../img/x-icon.png";
import githubIcon from "../img/github-icon.png";
import "../styles/contact.css";

const Contact = () => {
  return (
    <div className="contact">
      <Link className="contact__link" to="mailto:camillaprieststephens@gmail.com">
        <img className="contact__icon" src={emailIcon} alt="email icon" />
      </Link>
      <Link
        to="https://www.linkedin.com/in/camilla-priest-stephens/"
        target="_blank"
      >
        <img className="contact__icon" src={linkedInIcon} alt="LinkedIn icon" />
      </Link>
      <Link to="https://twitter.com/Camilla_Codes" target="_blank">
        <img className="contact__icon" src={xIcon} alt="X icon" />
      </Link>
      <Link to="https://github.com/cprieststephens" target="_blank">
        <img className="contact__icon" src={githubIcon} alt="Github icon" />
      </Link>
    </div>
  );
};

export default Contact;
