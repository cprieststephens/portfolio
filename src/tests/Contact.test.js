import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Contact from "../components/Contact";

describe("Contact", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct images", () => {
    render(
      <MemoryRouter>
        <Contact />
      </MemoryRouter>,
    );
    const emailIcon = screen.getByAltText("email icon");
    const linkedInIcon = screen.getByAltText("LinkedIn icon");
    const xIcon = screen.getByAltText("X icon");
    const gitHubIcon = screen.getByAltText("Github icon");

    expect(emailIcon).toBeInTheDocument();
    expect(linkedInIcon).toBeInTheDocument();
    expect(xIcon).toBeInTheDocument();
    expect(gitHubIcon).toBeInTheDocument();
  });
});
