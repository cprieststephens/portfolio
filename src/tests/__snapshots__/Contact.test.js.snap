// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<DocumentFragment>
  <div
    class="contact"
  >
    <a
      class="contact__link"
      href="mailto:camillaprieststephens@gmail.com"
    >
      <img
        alt="email icon"
        class="contact__icon"
        src="email-icon.png"
      />
    </a>
    <a
      href="https://www.linkedin.com/in/camilla-priest-stephens/"
      target="_blank"
    >
      <img
        alt="LinkedIn icon"
        class="contact__icon"
        src="linkedin-icon.png"
      />
    </a>
    <a
      href="https://twitter.com/Camilla_Codes"
      target="_blank"
    >
      <img
        alt="X icon"
        class="contact__icon"
        src="x-icon.png"
      />
    </a>
    <a
      href="https://github.com/cprieststephens"
      target="_blank"
    >
      <img
        alt="Github icon"
        class="contact__icon"
        src="github-icon.png"
      />
    </a>
  </div>
</DocumentFragment>
`;
